### ISSUE TITLE
Render the docs search bar on all branches

### ISSUE THREAD

**Issue by brookslybrand** [remix-run/remix-website#200]
## tl;dr

Render the docs search bar on all docs (`main`, `dev`, previous versions, etc.)

## Context

Recently I shared [a tweet](https://twitter.com/BrooksLybrand/status/1758601575000375496) hyping up some changes coming to the next version of Remix.

This highlighted the fact that switching between branches causes the search bar to disappear.

[I summarized the reason and the remediation options](https://twitter.com/BrooksLybrand/status/1759696092986290303) as follows

> It's happening because we only render the search box for `main` branch. We only render it for `main` because that's the only docs we index.

> So either we need to:
> a) index other branches
> b) just show the search box on every branch, even though it's only searching main

For the sake of consistency between branches, let's go with **b** for now. If we find that this causes confusion or frustration for users we can further enhance it with some sort of message or by implementing **a**
